src/build/build_helpers.jl

```diff
@@ -142,12 +142,18 @@ function get_expression_for_commodity_producer_price(jm, pf, commodity::Commodit
         for output in pf.outputs
             if output.commodity == commodity
                 for tax in output.taxes
-                    push!(taxes, tax.rate)
+                    push!(taxes, eval(swap_our_param_with_val(tax.rate)))
                 end
             end
         end
-
-    return tojump(jm, commodity) * -(1., taxes...)
+        if taxes == []
+            final_tax = (1- +(0., taxes...))
+        elseif sum(taxes...) == 0.
+            final_tax = -1.
+        else
+            final_tax = -(1., taxes...)
+        end
+    return tojump(jm, commodity) * final_tax
```

Comment:
```suggestion
    return tojump(jm, commodity) * (1 - +(0., taxes...))
```

The whole `if` block should go, and we just use that one line.